extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/MongoSourceBuilder.java

```diff
@@ -254,15 +256,26 @@ private Batch(
         }
 
         /**
-         * If {@code true}, the lack of database or collection will cause an error.
-         * If {@code false}, database and collection will be automatically created.
-         * Default value is {@code true}.
+         * If non {@link ResourceChecks#NEVER}, the lack of database or collection will cause an error.
+         * Otherwise, database and collection will be automatically created.
+         * Default value is {@link ResourceChecks#ONCE_PER_JOB}.
          *
-         * @param throwOnNonExisting if exception should be thrown when database or collection does not exist.
+         * @param checkResourceExistence if exception should be thrown when database or collection does not exist.
          */
         @Nonnull
-        public Batch<T> throwOnNonExisting(boolean throwOnNonExisting) {
-            params.setThrowOnNonExisting(throwOnNonExisting);
+        public Batch<T> checkResourceExistence(ResourceChecks checkResourceExistence) {
+            existenceChecks = checkResourceExistence;
+            return this;
+        }
+
+        /**
+         * If set to true, reading will be done in only one thread.
```

Comment:
```suggestion
         * If set to true, reading will be done in only one thread cluster-wide.
```

Reply:
btw, maybe the method should be named `forceReadTotalParallelismOne`?

Reply:
7831956e3b2ce0787afc566a54b89fa469b724b0